### ISSUE TITLE
Html2React passes the `classname` down to custom web components if the element has a `style` attribute 

### ISSUE THREAD

**Issue by michalczaplinski** [frontity/frontity#742]
**Expected behavior**

When using Web Components in React (e.g. `amp-img`, `amp-iframe`) we have to pass a `class` prop to the component instead of `className` like in regular react components. Currently emotion has a known bug where you cannot use the `css` prop on the Web Components because of this. See: https://github.com/emotion-js/emotion/issues/1071 or https://github.com/emotion-js/emotion/pull/1089.

In our the parse function in html2react, if a component has a `style` attribute, we pass the css prop to that component: https://github.com/frontity/frontity/blob/b21c48757b45904c4c13affb35e8dfbc7daf51c8/packages/html2react/src/libraries/parse/index.ts#L43-L45

However, due to the aforementioned emotion bug, this actually passes a classname prop to the DOM if the component is a Web Component.

**Steps involved to reproduce the problem**

I've created a failing test case in #741.

**Possible solution**

I think we can check if the component is a Web Component by using the same function that React uses to check it: https://github.com/facebook/react/blob/c954efa70f44a44be9c33c60c57f87bea6f40a10/packages/react-dom/src/shared/isCustomComponent.js

Then, if the component is a Web Component, we can create the class name using `css` from `@emotion/css` and pass it directly to the class prop: 

```js
// src/libraries/parse/index.ts

import { css as cssFunc } from '@emotion/css';

// inside of adaptNode function
if (isWebComponent()) {
  props.class = css(value);
} else {
  props.css = cssFunc(value);
}
```

**Comment by luisherranz** [comment]
I liked the proposed solution, but I would maintain `node.props.css` in the processor API  even for custom elements, because I think it will be confusing for the final users if not.

I would do the switch from `className` to `class` at the very last moment, when we pass the component to React.

I've made a video to explain it better: https://www.loom.com/share/1cc04a54e63e47409ae036c0854d43d9

This is my proposed change:

```ts
const handleNode = ({
  node,
  index,
  ...payload
}: HandleNodeParams): React.ReactNode => {
  // ....

  if (node.type === "element") {
    const { className, ...props } = node.props;

    // Populate either className or class, depending
    // if the component is a custom component.
    if (isCustomComponent(node.component)) props.class = className;
    else props.className = className;

    return (
      <node.component {...{ key: index, ...props }}>
        {node.children
          ? handleNodes({ nodes: node.children, ...payload })
          : null}
      </node.component>
    );
  }
};
```

**Comment by michalczaplinski** [comment]
I just quickly threw something together last night, but I agree that this is a better solution. Thanks @luisherranz I will try it 🙂

**Comment by michalczaplinski** [comment]
To clarify: I proposed to use the `css` function from `@emotion/css` because it returns a hashed class name (which is what I wanted). The one from `@emotion/react` returns an object that you pass to the `css` prop on the component like:

```js
const styles = css`height: auto;`; // styles is actually an object.

<div css={styles} />`
```

But in your approach I don't need to mess with `@emotion/css` of course, so it's much better :)

**Comment by michalczaplinski** [comment]
I've tried the above approach but unfortunately it does not work as we expected.

I think that the key thing to realize here is that the contents of the css prop are handled by emotion **in the babel plugin** which magically creates the class names from the content of the `css` prop.

So, when we get to the `handleNode()` function and are about to create the actual react components emotion still has not created the `className` prop, because it does so through babel. And as we know, emotion does not handle custom components so this is why it passes `className` instead of `class` even if the component is a custom component.
	
However, I have found a better approach to solve this issue which does not require installing `@emotion/css`.

I explain in more details in a video: https://www.loom.com/share/fd3e727cdb314fe1b1aa09c4807043ce
